src/realm.h

```diff
@@ -3801,6 +3801,18 @@ RLM_API void realm_sync_session_wait_for_upload_completion(realm_sync_session_t*
 RLM_API void realm_sync_session_handle_error_for_testing(const realm_sync_session_t* session, int error_code,
                                                          int category, const char* error_message, bool is_fatal);
 
+/**
+ * Wrapper for SyncSession::shutdown_and_wait + delete realm files.
+ * The realm files can/cannot be deleted. SyncSession::shutdown_and_wait will block until
+ * the sync session does not get closed.
+ * @param session ptr to a valid sync session
+ * @param realm_path a null terminated string containing the path where the realm files are
+ * @param did_delete a boolean flag indicating whether the files were deleted or not
+ * @return False in case of errors, otherwise returns True.
+ */
+RLM_API bool realm_sync_shutdown_and_delete_realm(const realm_sync_session_t* session, const char* realm_path,
```

Comment:
We should be able to find out the realm path on disk from the session itself, it shouldn't be a separate argument.